Create version 1

darwyncook opened the pull request.

Commit:
Create version 1

MPDean commented:
What works:
-InavlidEmail is raised when there is no '@' 
-InvalidSocial is raised when there are no '-'

What doesn't work:
-InvalidEmail is not raised when the first character in the user is a number
-InvalidEmail is not raised when there is punctuation or strange symbols in the user
-InvalidEmail is not raised when there are 2 '@' characters
-InvalidEmail is not raised when there is punctuation or strange symbols (other than '.') in the domain
-InvalidEmail is not raised when things other than 'com', 'gov', 'edu', and 'biz' is the final part in the domain (ex: 'net')
-InvalidEmail is not raised when the first character in the domain is a number
-InvalidEmail is not raised when a number is at the end of the gtld (ex: 'edu1')

-InvalidSocial is not raised when the ss# is zeroes (ex: 000-00-0000)
-InvalidSocial is not raised when the ss# is '078-05-1120'
-InvalidSocial is not raised when the number of the beast is included
-InvalidSocial is not raised when there aren't enough characters (ex: 12-34-5678)
-InvalidSocial is not raised when there are too many characters (ex: 1234-56-7890)
-InvalidSocial is not raised when there are letters included in the ss#


I haven't finished testing my hashing code yet

Commit:
Create v2.py

Is this better?

MPDean commented:
You've actually made it worse, so no.

darwyncook commented:
How about this?

import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass

"""
class Email:

    def __init__(self, email):
        try:
            a, b = email.split("@")
            z = b.split(".")
            if z[-1] in ["com", "net", "edu"]:
                self.email = email
            else:
                raise InvalidEmail
        except ImportError:
            raise InvalidEmail

    def __str__(self):
        return self.email
"""

class Email:

    def __init__(self, email):
        search_Obj = re.search(r'[a-zA-Z]([a-zA-Z0-9]*)@([a-zA-Z0-9]*)\.edu|biz|com|gov', email, flags=0)
        if search_Obj is not None:
            self.email = search_Obj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_Obj = re.search(r'[0-9]{3}-[0-9]{2}-[0-9]{4}', social, flags=0)
        if search_Obj is not None:
            self.social = search_Obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()

MPDean commented:
Email and SS still don't work.

darwyncook commented:
import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass


class Email:

    def __init__(self, email):
        searchobj = re.search(r'^[a-zA-Z]([a-zA-Z0-9]*)@([a-zA-Z0-9\.]*)\.(edu|biz|com|gov)$', email, flags=0)
        if searchobj is not None:
            self.email = searchobj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_obj = re.search(r'^(?!219-09-9999|078-05-1120)(?!666|000|9\d{2})\d{3}-(?!00)\d{2}-(?!0{4})\d{4}$',
                               social, flags=0)
        if search_obj is not None:
            self.social = search_obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()

darwyncook commented:
import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass


class Email:

    def __init__(self, email):
        searchobj = re.search(r'^[a-zA-Z]([a-zA-Z0-9]*)@[a-zA-Z]([a-zA-Z0-9\.]*)\.(edu|biz|com|gov)$', email, flags=0)
        if searchobj is not None:
            self.email = searchobj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_obj = re.search(r'^(?!219-09-9999|078-05-1120)(?!666|000|9\d{2})\d{3}-(?!00)\d{2}-(?!0{4})\d{4}$',
                               social, flags=0)
        if search_obj is not None:
            self.social = search_obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()
